godot-ffi/src/plugins.rs

```diff
@@ -32,6 +32,8 @@ macro_rules! plugin_execute_pre_main {
     ($body:expr) => {
         const _: () = {
             #[allow(non_upper_case_globals)]
+            // Triggered by $body.
+            #[allow(clippy::significant_drop_tightening)]
```

Comment:
The [lint docs](https://rust-lang.github.io/rust-clippy/stable/index.html#significant_drop_tightening) say:

> Searches for elements marked with `#[clippy::has_significant_drop]` that could be early dropped but are in fact dropped at the end of their scopes. In other words, enforces the “tightening” of their possible lifetimes.

And you mentioned this is caused by the guard. But are you sure we cannot apply this to the code that actually causes this?

https://github.com/godot-rust/gdext/blob/5470c111df970c4dbcfa0b5b0b1875fe0236dd99/godot-ffi/src/plugins.rs#L68-L70

If not, then `// Triggered by $body.` should at least be more descriptive. This isn't a common lint, and it has a very non-self-explanatory name.